code_route/tools/base.py

```
@@ -1,27 +1,292 @@
+            # Normalize result to ToolResult
+            if isinstance(result, ToolResult):
+                return result
+            elif isinstance(result, str):
+                # Check for error prefix (backwards compat)
+                if result.startswith("Error:"):
+                    return ToolResult.err(result[6:].strip())
+                return ToolResult.ok(result)
+            else:
+                return ToolResult.ok(str(result))
+
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)